Server/src/services/getBasicDataService.ts

```diff
@@ -0,0 +1,28 @@
+import { Connection, getConnection } from "typeorm";
+import { selectAllCategoriesQuery } from "../models/entities/Category";
+import { selectAllMaterialsQuery } from "../models/entities/Material";
+import { selectAllSubcategoriesQuery } from "../models/entities/Subcategory";
+import { ICategory } from "../models/interfaces/CategoryInterface";
+import { IMaterial } from "../models/interfaces/MaterialInterface";
+import { ISubcategory } from "../models/interfaces/SubcategoryInterface";
+
+export class retrieveBasicData {
+    private connection: Connection;

```

Comment:
We should try to keep things modular - a service shouldn't have access to a connection obj. Instead the connection should only be used by the model

Reply:
Makes sense; I've moved the connection obj to a dedicated model.